packages/frontend-2/components/project/page/settings/general/block/Delete/Delete.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div>
+    <ProjectPageSettingsBlock background title="Delete Project" :icon="TrashIcon">
+      <template #introduction>
+        <p class="font-bold mb-1 text-sm">
+          Deleting a project is an irreversible action.
+        </p>
+        <p>
+          If you are sure you want to proceed, click the button below to begin deletion.
+        </p>
+      </template>
+      <div class="flex justify-end w-full">
+        <div class="max-w-max">
+          <FormButton color="danger" outlined @click="showDeleteDialog = true">
+            Delete Project
+          </FormButton>
+        </div>
+      </div>
+    </ProjectPageSettingsBlock>
+    <ProjectPageSettingsGeneralBlockDeleteDialog
+      v-if="project"
+      v-model:open="showDeleteDialog"
+      :project="project"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { TrashIcon } from '@heroicons/vue/24/outline'
+import { graphql } from '~~/lib/common/generated/gql'
+import type { ProjectPageSettingsGeneralBlockDelete_ProjectFragment } from '~~/lib/common/generated/gql/graphql'
+
+graphql(`
+  fragment ProjectPageSettingsGeneralBlockDelete_Project on Project {
+    id
+    name
+    role
+    models {
```

Comment:
You should definitely use limit: 0 for both models & commentThreads, because
1. That way you access the same cache thet is used in the project page query, where these same params are used
2. The server does way less work, because it wont actually try to resolve any items